libsplinter/src/node_id/store/diesel/operations/get_node_id.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::super::{NodeID, NodeIdStoreError};
+use super::NodeIdOperator;
+use diesel::prelude::*;
+
+impl<'a, C> NodeIdOperator<'a, C>
```

Comment:
The pattern here is that the `get_node_id` function should be defined in a trait, and then implemented on the operations struct.

See https://github.com/Cargill/splinter/blob/main/libsplinter/src/registry/diesel/operations/get_node.rs as an example.